languages.adoc

```diff
@@ -18,88 +18,82 @@ SPDX-FileType: DOCUMENTATION
+A uProtocol Language specific libraries (_libraries_), are written for a given programming language (ex. Java, C++, etc...) to implement the protocol and are the primary library that is used by applications and services (uEs) to communicate with other uEs. The _libraries_ declare the API for the uProtocol Transport Layer (uP-L1) and implement the uProtocol Communication Layer API (uP-L2).
 
-NOTE: All libraries *MUST* be made available under a permissive (OSS) license like Apache 2.0 or MIT and *MUST* be hosted under the https://github.com/eclipse-uprotocol[Eclipse uProtocol GitHub Organization].
 
-== Source Code Structure
+== Requirements
 
-The following sections define the content and structure of these libraries.
+=== Content
 
-* The root namespace (package name) *MUST* be `org.eclipse.uprotocol`
-* Modules defined in table <<content>> below *MUST* be implemented in their own folder (namespace)
-* Each module *SHOULD* separate the implementation of the specification as defined in table <<folders>> below.
+[oft-sid="req~up-language-api~1"]
+--
+* *MUST* implement serializers, validators, and builders for the link:../basics/README.adoc[uProtocol data types] using the link:../up-core-api/README.adoc[up-core-api data model].
+--
 
+[oft-sid="req~up-language-transport-api~1"]
+--
+* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
+--
 
-.Module Folders/Namespace
-[#folders,width="80%",cols="20%,80%",options="header"]
-|===
-| Folder | Description
+[oft-sid="req~up-language-comm-api~1"]
+--
+* *MUST* provide an implementation of the link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) `API].
+--
 
-| `validators`
-| static methods to validate the data model
+[oft-sid="req~up-language-naming~1"]
+--
+* *MUST* be named according to pattern `up-[LANGUAGE]`
+--
 
-| `serializer`
-| Any code that is responsible to Serialize and deserialize the data model
+[oft-sid="req~up-language-structure~1"]
+--
+* The top level namespace (package name) *MUST* contain `uprotocol` and *SHOULD* contain `org.eclipse.uprotocol` 
+--
 
-| `factory` (or `builder`)
-| Factory methods to build the data model per business logic (i.e. to build CloudEvents, UUIDs, etc..)
+[oft-sid="req~up-language-documentation~1"]
+--
+ * *MUST* contain a `README.adoc` that explains how to build and use the library. 
```

Comment:
what about API docs?

Reply:
API documentation is in the code and not in the README usually.